### DISCUSSION TITLE
Visibility of Changes in the PDF

### DISCUSSION THREAD

**Discussion by Aadil-5122** [pymupdf/PyMuPDF#2226] [Looking for help] [answered]
Good Morning sir,

i would like to ask, if the changes in the PDF are visible without saving the PDF,

For example:

If i delete an image in a doc (say doc1), now **"without"** saving the doc, 
if i apply page.get_pixmap(clip=rect), where rect is the rectangle in which the image was contained, will i get the pixmap with respect to thenew state of the document (the one without the image)

Note that i havent saved the document in any step after making changes!

**Comment by JorjMcKie** [comment, marked as answer]
Yes, this will reflect changes made so far ... **_however_**  some changes, especially pertaining to annotations (new or updated), require that you "refresh" the page before taking a photo via `get_picmap()`. This is done by `page = doc.reload_page(page)`.

**Reply by Aadil-5122** [reply]
After deleting the "image" in a PDF (using the Delete image script in your repository), i reloaded the page, now when i try to use "get_pixmap" function, an error pops in:

pix= page.get_pixmap(clip=rectBack, alpha=False, dpi=150)
ValueError: orphaned object: parent is None

**Reply by Aadil-5122** [reply]
How do i resolve this?

**Reply by JorjMcKie** [reply]
You must use the full statement as I wrote it. Not just `doc.reload_page(page)`.

**Comment by Aadil-5122** [comment]
Yes Sir, Got it, Apologies for the inconveniences!
Thanks alot!

**Reply by JorjMcKie** [reply]
No problem 😉
